libtsuba/src/ParquetReader.cpp

```diff
@@ -148,141 +120,251 @@ tsuba::ParquetReader::ReadFromUriSliced(const katana::Uri& uri) {
   }
 
   std::shared_ptr<arrow::Table> out;
-  auto read_result = reader->ReadRowGroups(row_groups, &out);
-  if (!read_result.ok()) {
-    return KATANA_ERROR(ErrorCode::ArrowError, "arrow error: {}", read_result);
-  }
+  KATANA_CHECKED(reader->ReadRowGroups(row_groups, &out));
+  return out->Slice(row_offset, last_row - first_row);
+}
 
-  out = out->Slice(row_offset, slice.length);
+class BlockedParquetReader {
+public:
+  static Result<std::unique_ptr<BlockedParquetReader>> Make(
```

Comment:
perhaps some documentation about the different file formats that this method supports reading (parquet or offsets)?